### DISCUSSION TITLE
Missing preview_url

### DISCUSSION THREAD

**Discussion by lucacomploi** [spotify-web-api-java/spotify-web-api-java#311] [Q&A]
# Missing preview_url for track
I am having trouble with some missing data from the library. As an example, fetching track data of a track with id="37ZJ0p5Jm13JPevGcx4SkF" returns null for field preview_url. But fetching data of the same track with rapidapi.com returns a working url as shown in the screenshots below:

Example with null-value using the library:
![image](https://user-images.githubusercontent.com/55790767/197409286-1362d76e-b62c-4b0d-94ac-ce149c594fe2.png)

Same example with data fetched from rapidapi.com:
![image](https://user-images.githubusercontent.com/55790767/197409296-38a374b5-0203-4ece-8b7e-d79ee65845ca.png)

I'm not sure if I'm doing something wrong or if this is a bug. Does somebody know more?

**Comment by dargmuesli** [comment]
Please make sure to use the latest version of this library and show the code used to make the request for people to be able to offer assistance.

**Reply by lucacomploi** [reply]
Hello, thanks for your answer.

I am using the newest version of the library (7.2.2):
![image](https://user-images.githubusercontent.com/55790767/197877598-14baefb2-7612-4911-8ac9-7252140a3508.png)

All Spotify requests are done through the following static methods, returning the relative object from this library:
![image](https://user-images.githubusercontent.com/55790767/197878287-d8f9706f-0fad-428a-844d-25efcf16dc39.png)

This specific request is done with the response object being directly a method parameter for this method, and as shown the value for 'previewUrl' is null, but it shouldn't, based on data from rapidapi.com:
![image](https://user-images.githubusercontent.com/55790767/197880304-6cf1746d-ad3d-46d2-8ba1-a9c2ab41ed48.png)

Thanks in advance for any advice or help ;-)

**Reply by dargmuesli** [reply]
You can post code with syntax highlighting by using backtickts (without the backslashes) as follows:
```
\```java
{your code here}
\```
```

That would help people test your code. Alternatively you can share a repository.

Regarding your issue: try the authorization code flow instead, the client credentials flow might make the difference.

**Reply by lucacomploi** [reply]
Hello @dargmuesli,

Thanks for your help. I tried everything in Postman, also with client credentials flow, and there the field preview_url is null for this track too, no matter if I use simple login or the authorization code flow. To me, it seems like a Spotify problem.
